### PR TITLE
[README] Translate heading references

### PR DESCRIPTION
**Pull request by jablko**
I haven't found a way in GFM to specify section anchors separately from their heading text, which means when headings are translated, their incoming links need to be correspondingly updated.

This PR translates a few outstanding references, and helps authors spot them in future by adding a check to the CI.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongoose ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongoose@latest/))
was missing the following properties:
1. modelSchemas
1. options
1. cast
1. driver
1. ObjectId

<details>
<summary>as well as these 7 other properties...</summary>
<p>Decimal128, Mixed, Date, Number, now, CastError, SchemaTypeOptions</p>
</details>

## uikit ([<kbd>unpkg</kbd>](https://unpkg.com/browse/uikit@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'uikit'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. util
1. data
1. prefix
1. options

<details>
<summary>as well as these 1 other properties...</summary>
<p>version</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 33871b70097bc601e8fd42489bbd437d0a6ef1b7
</p>

**Comment by sheetalkamat** [comment]
Added @orta to review translations
@elibarzilay do you think its ok to add dependency on package that this PR adds.

File: .github/workflows/lint-md.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Check Markdown links
+on:
+  pull_request:
+    paths: "**.md"
+jobs:
+  lint-md:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: npm install
```
Comment (amcasey) [inline .github/workflows/lint-md.yml:10]: What does this install and why is it needed?

File: .github/workflows/lint-md.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+      - run: npx remark --use validate-links --frail . .github
```
Comment (amcasey) [inline .github/workflows/lint-md.yml:11]: I'm a little nervous about this hidden dependency.  It seems like it would be harder to overlook if it were declared in package.json.

Having said that, I think @orta already has some code for detecting dead links and I'm not sure this is consistent with the approach we use elsewhere.

File: types/uikit/README.md [hunk 3]
```diff
@@ -1,61 +0,0 @@
-[![uikit banner](https://cloud.githubusercontent.com/assets/321047/21769911/474d7d9e-d681-11e6-9fe0-d95f8ccfd3a9.jpg)](http://getuikit.com/)
```
Comment (amcasey) [inline types/uikit/README.md:1]: Are these deletions related to the rest of the change?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-md.yml (hunk 1 above)
Comment (jablko) [reply .github/workflows/lint-md.yml:10]: Installs the usual package dependencies, to which this change adds `remark-cli` and `remark-validate-links` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fe72bd3fdc8af7c6482de1819497913ba0ee1a04/package.json#L33-L34

File: .github/workflows/lint-md.yml (hunk 2 above)
Comment (jablko) [reply .github/workflows/lint-md.yml:11]: > I'm a little nervous about this hidden dependency. It seems like it would be harder to overlook if it were declared in package.json.

This change does declare those dependencies in `package.json`, if I understand your comment correctly? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fe72bd3fdc8af7c6482de1819497913ba0ee1a04/package.json#L33-L34

> Having said that, I think @orta already has some code for detecting dead links and I'm not sure this is consistent with the approach we use elsewhere.

I'm all in favor of an existing solution :+1:

File: types/uikit/README.md (hunk 3 above)
Comment (jablko) [reply types/uikit/README.md:1]: Yeah, they fail the new check https://github.com/DefinitelyTyped/DefinitelyTyped/runs/1420626076#step:6:73

because they https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/uikit/README.md

link to non-existent files, e.g. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/uikit/CONTRIBUTING.md

I think what's going on is that the README is copied from the underlying package https://github.com/uikit/uikit/blob/develop/README.md

without the ancillary files. I opted to remove rather than fix it because it documents the underlying package vs. the types.

Comment (amcasey) [reply types/uikit/README.md:1]: Makes sense.

File: .github/workflows/lint-md.yml (hunk 1 above)
Comment (amcasey) [reply .github/workflows/lint-md.yml:10]: I'm blind.  Thanks!

**Review by amcasey — COMMENTED** [review]
You've addressed my concerns, but I'd like to hear from @orta (or maybe @elibarzilay?) about the new dependency and existing mechanisms for solving this problem.  Alternatively, if you want to split the infrastructure change into a separate PR, I'll sign off on the link fixes now.

**Comment by elibarzilay** [comment]
* The added remark dependency seems ok, and I also like validating readmes, but IIUC this is a global test rather than testing just edited/added content which would be better if it's possible to do.
* If @orta has something existing then I'm on the same side too of preferring it
* It looks iffy to me to remove the failing MDs -- if it's not easy to fix them, maybe just edit them to make the failing links into some `FIXME` text for owners to fix?
* And re the translated `README.md` headers -- that makes it sound more tempting to avoid translations in the first place (re @sheetalkamat's PR)

**Comment by jablko** [comment]
> The added remark dependency seems ok, and I also like validating readmes, but IIUC this is a global test rather than testing just edited/added content which would be better if it's possible to do.

I recently revised this so it only triggers when a Markdown file is edited, and given that this excludes the majority of PRs, I'm not sure about the benefit of revising again to only check the affected files? As long as we fix the existing failures, which I think is (minorly) beneficial ...

> It looks iffy to me to remove the failing MDs -- if it's not easy to fix them, maybe just edit them to make the failing links into some FIXME text for owners to fix?

I think those two files should be removed regardless: They belong in the underlying package's repo (where they're found already); I think READMEs in the DT repo should mention the types (which these don't). But maybe it isn't very polite to remove them as part of this PR ... I'll open a separate PR to ask the authors if they can be removed: #49932

**Comment by elibarzilay** [comment]
> But maybe it isn't very polite to remove them as part of this PR ... I'll open a separate PR to ask the authors if they can be removed: #49932

(Yeah -- I don't have an opinion, I just wanted to ensure that they're aware of the removal...)

**Comment by jablko** [comment]
:alarm_clock::zzz:

**Review by amcasey — APPROVED** [review]
I like the fixes and I have yet to hear any concrete objections to the new process.

File: .github/workflows/lint-md.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+name: Check Markdown links
+on:
+  pull_request:
+    paths: "**.md"
```
Comment (orta) [inline .github/workflows/lint-md.yml:4]: TIL'd

File: .github/workflows/lint-md.yml (hunk 2 above)
Comment (orta) [reply .github/workflows/lint-md.yml:11]: I have an offline one for the TS website, but I think this is a good strategy

**Review by orta — APPROVED** [review]
Looks like a good idea to me

**Comment by jablko** [comment]
Ready to merge, thanks!

**Comment by andrewbranch** [comment]
@typescript-bot ya snoozin? ☝️
